oak_functions_launcher/src/lib.rs

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#![feature(never_type)]
+#![feature(result_flattening)]
+#![feature(array_chunks)]
+
+use crate::channel::ConnectorHandle;
+use anyhow::Context;
+use instance::{native, virtualized, LaunchedInstance};
+use std::{
+    fs,
+    io::{BufRead, BufReader},
+    os::unix::net::UnixStream,
+    path::PathBuf,
+};
+
+use crate::schema::InitializeResponse;
+
+pub mod schema {
+    #![allow(dead_code)]
+    use prost::Message;
+    include!(concat!(env!("OUT_DIR"), "/oak.functions.rs"));
+}
+
+mod channel;
+pub mod instance;
+mod lookup;
+pub mod server;
+
+#[derive(clap::Subcommand, Clone, Debug, PartialEq)]
+pub enum Mode {
+    /// Launch a virtual enclave binary
+    Virtual(virtualized::Params),
+    /// Launch an enclave binary directly as a child process
+    Native(native::Params),
+}
+
+pub async fn create(
+    mode: Mode,
+    lookup_data_path: PathBuf,
+    wasm_path: PathBuf,
+    constant_response_size: u32,
+) -> Result<
+    (
+        Box<dyn LaunchedInstance>,
+        ConnectorHandle,
+        InitializeResponse,
+    ),
+    Box<dyn std::error::Error>,
+> {
+    let (launched_instance, connector_handle) = launch_instance(mode).await?;
+    setup_lookup_data(connector_handle.clone(), lookup_data_path).await;
```

Comment:
The fact that we're cloning this handle makes me rather uncomfortable. Semantically, they should borrow the handle (as it would make no sense for them to use it later).